manifests/cluster-infra/6b_loki_stateful_set.yaml

```diff
@@ -0,0 +1,140 @@
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: loki
+  labels:
+    app: loki
+data:
+  loki.yaml: |
+    auth_enabled: false
+    ingester:
+      chunk_idle_period: 15m
+      chunk_block_size: 262144
+      lifecycler:
+        ring:
+          kvstore:
+            store: inmemory
+          replication_factor: 1
+    limits_config:
+      enforce_metric_name: false
+      reject_old_samples: true
+      reject_old_samples_max_age: 168h
+    schema_config:
+      configs:
+      - from: 2018-04-15
+        store: boltdb
+        object_store: filesystem
+        schema: v9
+        index:
+          prefix: index_
+          period: 168h
+    server:
+      http_listen_port: 3100
+    storage_config:
+      boltdb:
+        directory: /data/loki/index
+      filesystem:
+        directory: /data/loki/chunks
+    chunk_store_config:
+      max_look_back_period: 0
+    table_manager:
+      retention_deletes_enabled: false
+      retention_period: 0
+---
+apiVersion: v1
+kind: Service
+metadata:
+  name: loki-headless
+  labels:
+    app: loki
+spec:
+  clusterIP: None
+  ports:
+    - port: 3100
+      protocol: TCP
+      name: http-metrics
+      targetPort: http-metrics
+  selector:
+    app: loki
+---
+apiVersion: apps/v1
+kind: StatefulSet
```

Comment:
why do we need it as a stateful set?

Reply:
It's not needed but I think it's some kind of best practice to use statefulset when using persistent storage. If we follow that it'll be ideal to migrate `prometheus-meta` to `StatefulSet` too at some point. I've noticed it in some other repository too.

https://github.com/kubernetes/kubernetes/blob/master/cluster/addons/prometheus/prometheus-statefulset.yaml

But  there's an issue with wait time in prombench for `StatefulSet` which needs more investigation.

Should I change this to Deployment? (The actual loki helm charts are using `StatefulSet`)